### PR TITLE
Create Monad Instance for Try type

### LINKED ISSUE #44: Create Monad Instances for Try Type

### PR DESCRIPTION
**Pull request by JMPergar**
Fixes #44

### CODE REVIEW

File: katz/src/main/kotlin/katz/data/Try.kt [hunk 1]
```diff
@@ -22,7 +22,9 @@ package katz
  *
  * Port of https://github.com/scala/scala/blob/v2.12.1/src/library/scala/util/Try.scala
  */
-sealed class Try<out A> {
+sealed class Try<out A> : HK<Try.F, A> {
```
Comment (ffgiraldez) [inline katz/src/main/kotlin/katz/data/Try.kt]: create a typealias to improve readability inside Monads
``` kotlin
typealias TryKind<A> = HK<Try.F, A>
```

to improve read
```kotlin
override fun <A, B> map(fa: TryKind<A>, f: (A) -> B): Try<B> =  fa.ev().map(f)
```

File: katz/src/main/kotlin/katz/instances/TryMonad.kt [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package katz
+
+object TryMonad : Monad<Try.F> {
```
Comment (raulraja) [inline katz/src/main/kotlin/katz/instances/TryMonad.kt]: This type should not implement Monad but MonadError<F, Throwable>

File: katz/src/main/kotlin/katz/instances/TryMonad.kt [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package katz
+
+object TryMonad : MonadError<Try.F, Throwable> {
```
Comment (raulraja) [inline katz/src/main/kotlin/katz/instances/TryMonad.kt]: :clap:

**Comment by ffgiraldez** [comment]
Test are gonna be later?

**Pull request merged**